sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TransmitterFactory.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    /// <summary>
+    /// This Factory encapsulates the <see cref="AzureMonitorTransmitter"/>.
+    /// An ideal users will create a single exporter for each signal (Logs, Metrics, Traces).
+    /// This factory should ensure that only one instance of the Transmitter is created for
+    /// any unique connection string.
+    /// </summary>
+    internal class TransmitterFactory
+    {
+        public static TransmitterFactory Instance = new();
+
+        internal readonly Dictionary<string, AzureMonitorTransmitter> _transmitters = new();
+        private readonly object _lockObj = new();
+
+        public AzureMonitorTransmitter Get(AzureMonitorExporterOptions azureMonitorExporterOptions, TokenCredential? tokenCredential = null)
+        {
+            var key = azureMonitorExporterOptions.ConnectionString ?? string.Empty;
+
+            if (!_transmitters.TryGetValue(key, out AzureMonitorTransmitter transmitter))
+            {
+                lock (_lockObj)
```

Comment:
If ConcurrentDictionary could avoid this lock, my vote will be for it.

Reply:
see the other thread above.